way_to_home/static/src/components/routeForm/routeForm.js

```diff
@@ -20,9 +21,35 @@ class RouteSearchForm extends Component {
     }));
   }
 
-   getCurrentPosition = () => {
-     alert('get and write current position')
-   }
+   getCurrentPosition = (props) => {
+      if (!navigator.geolocation){
+        console.log("Geolocation is not supported by your browser")
+        return;
+      }
+      self = this
+      function success(position) {
+        let latitude  = position.coords.latitude;
+        let longitude = position.coords.longitude;
+        let url = 'https://nominatim.openstreetmap.org/reverse?format=jsonv2&lat='+latitude+'&lon='+longitude
+         axios.get(url)
+            .then(function (response) {
+                let addr = response.data.address.city + ', ' + response.data.address.road;
+                if (response.data.address.house_number){
+                    addr += ", " + response.data.address.house_number;
+                }
+                props == 'A' ? self.setPointA(addr) :  self.setPointB(addr);
+            })
+            .catch(function (error) {
+                throw error;
+            });
+        }
+
+      function error() {
+         throw "Unable to retrieve your location";
```

Comment:
Therefore, the user needs to go to the console to find out that the system can not get its geolocation. I think it is not the best decision.

Reply:
my mistake